src/index.js

```diff
@@ -193,6 +193,12 @@ function isTabbableRadio(node) {
 function isHidden(node) {
   if (getComputedStyle(node).visibility === 'hidden') return true;
 
+  const isDirectSummary = node.matches('details>summary:first-of-type');
+  const nodeUnderDetails = isDirectSummary ? node.parentElement : node;
+  if (nodeUnderDetails.matches('details:not([open]) *')) {
+    return true;
+  }
```

Comment:
Would it make sense to move this into an IF block?
```javascript
  if (node.matches('details>summary:first-of-type')) {
    return node.parentElement.matches('details:not([open]) *');
  }
```

In any case, based on samples in https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details, it seems `<details>` elements on their own, even if not open, are tabbable, whether a `<summary>` exists or not inside of it.

My impression is we aren't considering that case here. WDYT?

It seems it's actually the `<details>` element that's tabbable, not the `<summary>`. The later just affects that gets displayed in the toggle, whether it's just "Details" (default without summary) or it's some other text (specified in the summary)...

Reply:
> Would it make sense to move this into an IF block?

It would have to be 2 IF blocks with duplicate selector
```
if (
    node.matches('details>summary:first-of-type') &&
    node.parentElement.matches('details:not([open]) *')
  ) {
    return true;
  } else if (node.matches('details:not([open]) *')) {
    return true;
  }
```

> `<details>` elements on their own, even if not open, are tabbable

Good catch! From what I can see the actual `activeElement` for a details element with no summary is the details element itself, while the summary element get the focus if it exist (calling `focus()` on the details element will move the focus to the summary element).

I added details elements with no summary to be picked up as tabbable.

Reply:
Thanks for the JSBin, very helpful! Cool, so we found something more to fix! 🎉 

About the IF statement, though, I'm still wondering if we could optimize this. It would be nice if we didn't have to test every node that gets passed to this function as being a summary element, and then the other query.

Could we not do this instead?

```javascript
function isHidden(node) {
  if (getComputedStyle(node).visibility === 'hidden') return true;

  // 👉 isn't this sufficient to see if it's a node that hidden under a collapsed <details> element?
  if (node.matches('details:not([open]) *')) {
    return true;
  }

  while (node) {
    if (getComputedStyle(node).display === 'none') return true;
    node = node.parentElement;
  }

  return false;
}
```

Otherwise, it's a `<details>` element itself, and if it's hidden, then the first check would get it, or the first check in the `while` loop below. Or am I missing something subtle?

Reply:
This checks 2 options

- `direct first summary`
    1. check that it's actually is the first summary of a details element (which will make it focusable even when it's parent details is not open)
    2. check that the **summary's details (it's parent)** isn't deeply nested in a closed details element
- `any other element` - just check that the element isn't deeply nested in a closed details element

The code you suggested would find a `direct first summary` in a closed details as hidden:
```
if (node.matches('details:not([open]) *')) {
    return true;
 }
```

I could replace the first check with an imperative check (check type, check parent, check previous siblings), but I'm not sure it would be faster, and it would definitely be more code.

I could maybe make it a tiny bit faster by just checking the first condition (type), but again, that probably is just an insignificant performance optimization, and maybe even not...
```
const isDirectSummary = node.tagName === 'SUMMARY' && node.matches('details>summary:first-of-type');
// instead of
const isDirectSummary = node.matches('details>summary:first-of-type');
```

Reply:
Ah, OK. That's the detail I was missing in my head: the nesting. Thank you for being patient with me on this thread as I worked things out!